src/org/labkey/test/components/glassLibrary/components/react/typeahead/TypeAheadCombo.java

```diff
@@ -0,0 +1,136 @@
+package org.labkey.test.components.glassLibrary.components.react.typeahead;
+
+import org.labkey.test.Locator;
+import org.labkey.test.WebDriverWrapper;
+import org.labkey.test.components.Component;
+import org.labkey.test.components.WebDriverComponent;
+import org.labkey.test.selenium.EphemeralWebElement;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+import java.util.List;
+
+/**
+ *  Wraps the rbtCombo from react-bootstrap-typeahead; the client component is implemented in PropertyLookup.tsx
+ */
+public class TypeAheadCombo extends WebDriverComponent<TypeAheadCombo.ElementCache>
+{
+    final WebElement _el;
+    final WebDriver _driver;
+
+    public TypeAheadCombo(WebElement element, WebDriver driver)
+    {
+        _el = element;
+        _driver = driver;
+    }
+
+    @Override
+    public WebElement getComponentElement()
+    {
+        return _el;
+    }
+
+    @Override
+    public WebDriver getDriver()
+    {
+        return _driver;
+    }
+
+    public TypeAheadCombo setValue(String value)
+    {
+        if (hasValue())
+            clear();
+        elementCache().input.sendKeys( value);
+
+        elementCache().option(value).click();
+        WebDriverWrapper.waitFor(()-> !isOpened() && getValue().equals(value),
+                "the value was not set, or the combo did not close", 2000);
+        return this;
+    }
+
+    public String getValue()
+    {
+        return elementCache().input.getAttribute("value");
+    }
+
+    public boolean hasValue()
+    {
+        return !getValue().isEmpty();
+    }
+
+    // open
+    protected boolean isOpened()
+    {
+        return elementCache().input.getAttribute("aria-expanded").equals("true");
+    }
+
+    public TypeAheadCombo clear()
+    {
+        if (hasValue())
+            elementCache().clearBtn.click();
+
+        WebDriverWrapper.waitFor(()-> !hasValue(), "the value was not cleared in time", 2000);
+        return this;
+    }
+
+
```

Comment:
Is there a way to get the values that are in the drop down list of the combo? A security test for example may need to check that only specific values are shown in a list/combo control to users with limited permissions.

Reply:
They will appear as rendered dom elements in the flyout; should be relatively easy to implement